src/wiki_ulmfit.py

```diff
@@ -0,0 +1,99 @@
+import logging
+
+import gdown
+import sentencepiece as spm
+from fastai import *
+from fastai.text import *
+from gensim.corpora import WikiCorpus
+
+from src.utils import LangTokenizer
+
+logger = logging.getLogger()
+
+
+class WikiTrainer:
+    def __init__(self, path, lang="id"):
+        self.path = path
+        self.lang = lang
+
+    def download_wiki_files(self, wiki_text=False):
+        """
+        Downloads Processed Wikipedia from google drive. 
+        """
+        lang_dict = {
+            "id": {
+                "wiki": "https://drive.google.com/uc?export=download&id=10P95rNlMPuHyB57k40KFcZUlaXdFopUC",
+                "data": "https://drive.google.com/uc?export=download&id=1-0gaC0bmYyMoUWIkGoH6fY6DUQTArMD3",
+            }
+        }
+        if not (self.path / "data_save.pkl").exists():
+            gdown.download(
+                lang_dict[self.lang]["data"], f"{self.path}/data_save.pkl", quiet=False
+            )
+        if wiki_text:
+            gdown.download(
+                lang_dict[self.lang["wiki"]],
+                f"{self.path}/{self.lang}_wiki.txt",
+                quiet=False,
+            )
+
+    def load_data_lm(self):
```

Comment:
How does the API change if I want to use a finetuned LM instead of Wiki LM to predict text? 

Decouple the train, load and predict functions form "Wiki" models. It should be able to work with any fastai LM.

Reply:
Created `load_data_file` instead which will load any TextBunch given either the pickle file name or `.txt` file.